indico/modules/events/surveys/util.py

```diff
@@ -59,3 +67,46 @@ def was_survey_submitted(survey):
     if submission_id is None:
         return False
     return bool(SurveySubmission.find(id=submission_id).count())
+
+
+def generate_csv_from_survey(survey, submitter_ids):
+    """Generates a CSV file from a given survey.
+
+    :param submitter_ids: The list of `User` ids. Generated file includes only submissions made by those users
+    """
+    fieldnames = {'submitter', 'submission_date'}
+    fieldnames |= {'{}_{}'.format(question.title, question.id) for question in survey.questions
+                   if not isinstance(question.field, StaticTextField)}
+
+    with NamedTemporaryFile(suffix='indico.tmp', dir=Config.getInstance().getTempDir(), delete=False) as temp_file:
+        submissions = _filter_submissions(survey, submitter_ids)
+        writer = csv.DictWriter(temp_file, fieldnames=fieldnames)
+        writer.writeheader()
+        for submission in submissions:
+            submission_dict = {
+                'submitter': (submission.user.full_name.encode('utf-8')
+                              if submission.user else 'Anonymous'),
+                'submission_date': format_datetime(submission.submitted_dt),
+            }
+
+            for answer in submission.answers:
+                key = '{}_{}'.format(answer.question.title, answer.question.id)
+                submission_dict[key] = _prepare_data(answer.answer_data)
+            writer.writerow(submission_dict)
+        return temp_file
+
+
+def _filter_submissions(survey, submitter_ids):
+    submissions = SurveySubmission.find(SurveySubmission.survey == survey)
+    if not -1 in submitter_ids:
```

Comment:
It was just terrible wording. You did select submissions (also, it doesn't make any sense to select submitters since that way you'd lose the ability to select only some anonymous submissions)